lib/jobsub_api.py

```diff
@@ -74,3 +90,519 @@ def jobsub_call(argv: List[str], return_output: bool = False) -> Optional[str]:
         print(f"Excepion in jobsub_call({argv})")
         raise
     return res
+
+
+# =-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+
+def optfix(s: Optional[str]) -> str:
+    """typing fix -- make string not Optional"""
+    if not s:
+        return ""
+    return s
+
+
+# pylint: disable=too-many-instance-attributes
+class SubmittedJob(Job):
+    """result of fancier jobsub_submit call"""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        group: str,
+        jobid: str,
+        pool: str = "",
+        auth_methods: str = "",
+        role: str = "",
+        submit_out: str = "",
+    ) -> None:
+        """save various job submission values to reuse for hold, queue, etc."""
+        Job.__init__(self, jobid.strip())
+        self.group = group.strip()
+        self.pool = pool.strip()
+        self.auth_methods = auth_methods.strip()
+        self.role = role.strip()
+        self.submit_out = submit_out
+        self.status = None
+        self.set_q_attrs()
+
+    def set_q_attrs(
+        self,
+        owner: str = "",
+        submitted: str = "",
+        runtime: str = "",
+        status: str = "",
+        command: str = "",
+        prio: str = "",
+        size: str = "",
+    ) -> None:
+        """note values from jobsub_q output, owner, submitted, status, etc."""
+        init_JSMAP()
+        self.owner = owner
+        self.submitted = jsq_date_to_datetime(submitted) if submitted else None
+        self.runtime = jsq_runtime_to_timedelta(runtime) if runtime else None
+        self.status = JSMAP[status] if status else None
+        self.prio = float(prio) if prio else None
+        self.size = float(size) if size else None
+        self.command = command
+
+    def _update_args(self, verbose: int, args: List[str]) -> None:
+        """code adding common arguments to commands"""
+        args.append("-G")
+        args.append(self.group)
+        if verbose:
+            args.append("--verbose")
+            args.append(str(verbose))
+        if self.pool:
+            args.append("--global-pool")
+            args.append(self.pool)
+        if self.auth_methods:
+            args.append("--auth-methods")
+            args.append(self.auth_methods)
+        if self.role:
+            args.append("--role")
+            args.append(self.role)
+        args.append(self.id)
+
+    def hold(self, verbose: int = 0) -> str:
+        """Hold this job with jobsub_hold"""
+        args = ["jobsub_hold"]
+        self._update_args(verbose, args)
```

Comment:
Since these last three lines are run for hold, release, and rm, could they be combined into a single function, which gets called from each of those funcs?  Something like:

```python
def some_better_name(self, command, verbose):
    args = [command]
    self._update_args(verbose, args)
    rs = optfix(jobsub_call(args, True))
    return rs

def hold(self, verbose):
    return self.some_better_name("jobsub_hold", verbose)

def release(self, verbose):
    return self.some_better_name("jobsub_release", verbose)

def rm(self, verbose):
    return self.some_better_name("jobsub_rm", verbose)
```

Reply:
So that was the Idea of the _update_args routine already, so I collapsed it into one called _cmd() that takes
the initial args with the base command and  base options, and moved the optfix(jobsub_call...) up there.